src/lib.rs

```diff
@@ -456,16 +457,53 @@ impl SandboxFS {
     }
 }
 
+/// Creates a file `path` with the given `uid`/`gid` pair.
+///
+/// The file is created via the `create` lambda, which can create any type of file it wishes.  The
+/// `delete` lambda should match this creation and allow the deletion of the file, and this is used
+/// as a cleanup function when the ownership cannot be successfully changed.
+fn create_as<T, E, P, Create, Delete>(
+    path: &P, uid: unistd::Uid, gid: unistd::Gid, create: Create, delete: Delete) -> Result<T, E>
+    where P: AsRef<Path>, Create: Fn(&P) -> Result<T, E>, Delete: Fn(&P) -> Result<(), E>,
+          E: From<Errno> {
+    let mut res = create(path);
+    if res.is_ok() {
+        if let Err(e) = unistd::fchownat(None, path.as_ref(), Some(uid), Some(gid),
+                                         unistd::FchownatFlags::NoFollowSymlink) {
+            res = Err(E::from(e.as_errno().expect("fchownat should have failed with an errno")));
```

Comment:
Prefer to handle the weird instead of panicking.

Reply:
Dubious about this. I don't think any of the other non-errno errors in nix are possible here, so getting them would mean that our logic is wrong -- and in those cases I favor crashing than returning an error.

But I'm not fully sure, so done. Doing this... made me change the structure of this function significantly. Not sure if it's now easier to read or worse than before.